### PR TITLE
Muon workspace grouping

### LINKED ISSUE #32860: muon speed up
replace get_y_min_max_between_x_range in plotting_canvas_utils with 

```
    x, y = line.get_data()
    start = np.searchsorted(x, x_min)
    end = np.searchsorted(x, x_max)
    crop = y[start:end]
    y_min = np.min(crop)
    y_max = np.max(crop)
    return y_min, y_max
```

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**

The muon GUI was slow and this was mainly due to the overzealous calls to group workspace. This changes the code so it only calls group workspace at the end of the calculations. I have preserved the output in the ADS from previous versions. 

I have also added a small speed up in the plotting by changing the way autoscale works.

**To test speed ups**

On the main branch (release build) open muon analysis (MA) and load EMU 114150-60 and time how long it takes (on my machine its about 16 sec)

Get this branch and turn on the model fitting (see https://stfc365.sharepoint.com/:w:/r/sites/Mantid/_layouts/15/Doc.aspx?sourcedoc=%7B908F6EE2-0342-4843-90BE-7217122A9F08%7D&file=Feature%20flag%20instructions.docx&action=default&mobileredirect=true)

Open mantid
Open MA
load EMU 114150-60 and time how long it takes. This should be a lot faster than main.

Turn off autoscale and zoom in on the plot
Turn autoscale back on and check that works

**To test grouping is still correct**

Open MA
Load some data (e.g. MUSR 62260-1)
Check that all of the workspaces are in the correct groups (if you load N runs you should have N groups)

Change Rebin to fixed
Enter 10
Check the new workspaces have been added to the correct groups

Go to the grouping tab
Click the `+` below one of the tables and enter a name (e.g. test)
Check that the new workspaces are in the correct groups

Go to the phase table tab
Click `calculate phase table` and call it "table"
Check its added to the correct group (should only be in the group that matches the workspace on the first row of the table)
Click the `+` at the bottom and give it a  name (e.g. phase)
Check that the workspaces are added to the correct groups

Go to fitting
Add a function, click fit
You should get a new group with 3 workspaces (data, parameters and covarience matrix) 
Tick the simultaneous box, change the fit function (add a flat background) and do another fit
Another group should appear (for the suggested runs you will have 6 things in the group)

Go to sequential fit
Press `sequentially fit all`
Check you have the new group

Go to results and click `Output Results`
A table and a group (will include the word combinations) will appear in the ADS

Go to the Model fitting tab
Add a function and press fit
A new group will appear in the ADS

Close MA, clear the ADS

Open frequency domain analysis
Load MUSR 62263
Go to the transform tab
Click `calculate`
The workspaces should be added to the group
Change the drop down at the top of the tab from FFT to maxent
Tick all of the checkboxes in the top table
Click calculate - this is slow
When its done all of the workspaces should be added to the group

Fixes #32860. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
autoscale is disabled until you turn zoom off again

**Comment by sf1919** [comment]
> Another group should appear (for the suggested runs you will have 6 things in the group)

I ended up with 8 but looking at the groups it also seems to have included the PhaseQuads too. Is it supposed to do that?

Screenshot (61)

**Comment by AnthonyLim23** [comment]
If you have them selected in the phase tab then yes

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/contexts/muon_context.py [hunk 1]
```diff
@@ -371,36 +375,35 @@ def update_current_data(self):
...
+            MuonWorkspaceWrapper(loaded_workspace_deadtime_table).show(directory + deadtime_name)
+            self.corrections_context.set_default_dead_time_table_name_for_run(self.data_context.instrument, run,
+                                                                              deadtime_name)
+
+            if len(loaded_workspace) > 1:
+                # Multi-period data
+                for i, single_ws in enumerate(loaded_workspace):
                     name = directory + get_raw_data_workspace_name(self.data_context.instrument, run_string,
-                                                                   multi_period=False,
+                                                                   multi_period=True,
```
Comment (sf1919) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/contexts/muon_context.py:397]: Does it matter that multi_period has changed from False to True?

**Review by sf1919 — APPROVED** [review]
I have tested the PR on the Muon Analysis (MA) GUI, Frequency Domain Analysis GUI and even with the new Elemental Analysis GUI that is in development. It passed all the tests and functioned as expected. 

The only exception to this was how much quicker the loading in the MA GUI was on my machine. In the release build it ran for 14 seconds. This is definitly quicker than the 16 seconds previously but not as big a change as inferred in the PR. It will be interesting to see how much quicker the GUI is once this and PR #32875 are both merged in.

I have reviewed the code too. Good to see the new functions are being tested.

I am therefore happy to approve.

**Comment by sf1919** [comment]
@mantidproject/gatekeepers - could this be looked at soon? We're hoping to have this merged in ready for users to look at during an upcoming workshop.

**Pull request merged**